service-loadbalancer/service_loadbalancer.go

```diff
@@ -417,24 +438,78 @@ func (lbc *loadBalancerController) getServices() (httpSvc []service, tcpSvc []se
 				newSvc.FrontendPort = servicePort.Port
 				tcpSvc = append(tcpSvc, newSvc)
 			} else {
-				if val, ok := serviceAnnotations(s.ObjectMeta.Annotations).getCookieStickySession(); ok {
-					b, err := strconv.ParseBool(val)
-					if err == nil {
-						newSvc.CookieStickySession = b
-					}
-				}
-
 				newSvc.FrontendPort = lbc.httpPort
 				httpSvc = append(httpSvc, newSvc)
 			}
 			glog.Infof("Found service: %+v", newSvc)
 		}
+
+		sort.Sort(serviceByName(httpSvc))
+		sort.Sort(serviceByName(tcpSvc))
 	}
+	if lbc.useIngress {
+		// Overwrite the path/host rules if http services based on Ingress.
+		httpSvc = lbc.fillIngressRules(httpSvc)
+	}
+	return
+}
 
-	sort.Sort(serviceByName(httpSvc))
-	sort.Sort(serviceByName(tcpSvc))
+// fillIngressRules writes rules from the Ingress to the Service.
+// If no Ingress exists for a Service:port, it's discarded.
+func (lbc *loadBalancerController) fillIngressRules(svcs []service) []service {
+	svcLookup := map[string]service{}
+	for _, svc := range svcs {
+		svcLookup[svc.Name] = svc
+	}
+	out := []service{}
+	for _, m := range lbc.ingStore.List() {
+		ing := *m.(*extensions.Ingress)
+		for _, rules := range ing.Spec.Rules {
```

Comment:
This actually doesnt take into accout the default backend. If you don't want to modify this, you can add a todo. 

This is an Ingress with default backend:
https://github.com/kubernetes/kubernetes/blob/release-1.1/docs/user-guide/ingress.md#single-service-ingress